airflow/providers/openlineage/plugins/adapter.py

```diff
@@ -350,6 +358,7 @@ def _build_run(
     @staticmethod
     def _build_job(
         job_name: str,
+        job_type: JobTypeJobFacet,
```

Comment:
It doesn't need to be a parameter. Can't foresee any scenario in which OpenLineageAdapter from Airflow OL provider would send any other job type.

In that case it would be simpler just to change `_build_job` method to add the facet and in `dag_start/failed/success` use the method instead of creating the `Job` directly.

Reply:
Thanks for the suggestion 
I used the method `_build_job` to create the Job in the dag methods.

Reply:
Does it still need to be an argument for `_build_job`?

Reply:
I can remove it but I need to define two methods one `_build_job_dag` and the other `_build_job_task` and using the `_build_job` with `_JOB_TYPE_DAG` or  `_JOB_TYPE_TASK` passed as a argument.

Reply:
I think that approach is fine as far as we have the same code for `DagRun` and `TaskInstance` listener

Reply:
@mattiabertorello  you're right, sorry I missed it!